nemo_skills/inference/model/tool_call.py

```diff
@@ -113,33 +114,38 @@ async def generate_async(
         result_steps = defaultdict(list)
         conversation = copy.deepcopy(prompt)
 
+        # assigning a unique request id to pass to tool calls if they need to be stateful
+        request_id = str(uuid.uuid4())
+
         while True:
             if isinstance(tokens_to_generate, int) and tokens_to_generate <= 0:
                 break
-
             generation = await self.model.generate_async(
                 prompt=conversation,
                 tools=tools,
                 tokens_to_generate=tokens_to_generate,
                 **generation_kwargs,
             )
-
             if isinstance(tokens_to_generate, int):
                 tokens_to_generate -= generation["num_generated_tokens"]
 
             for k in ["generation", "num_generated_tokens", "reasoning_content", "finish_reason"]:
                 if k in generation:
                     result_steps[k].append(generation[k])
 
-            conversation.append({"role": "assistant", "content": result_steps["generation"][-1]})
+            conversation.append({"role": "assistant", "content": generation["generation"]})
+            if "reasoning_content" in generation:
```

Comment:
Just confirming, does this line align with the typical expectation, e.g. from [Qwen Models](https://qwen.readthedocs.io/en/latest/getting_started/concepts.html#thinking):

> The thinking block should only be included in the final round except for multi-step tool calls.

Reply:
I believe this is handled by their chat template. So we should always give full conversation history, but the model will strip it away in jinja